chord_metadata_service/discovery/utils.py

```diff
@@ -123,3 +131,44 @@ def get_discovery_entity_model_scoped_queryset(entity: DiscoveryEntity, scope: V
     Small utility for the semi-common usage pattern of getting a scoped queryset for a discovery entity's Django model.
     """
     return DISCOVERY_ENTITY_NAMES_TO_MODEL[entity].get_model_scoped_queryset(scope)
+
+
+def get_entity_counts_for_scope(scope: ValidatedDiscoveryScope) -> EntityCounts:
+    """
+    Returns entity counts for all discovery entities within the given scope.
+    Uses distinct() for entities that can appear multiple times through relationships.
+    """
+    return {
+        entity: get_discovery_entity_model_scoped_queryset(entity, scope).distinct().count()
+        for entity in DISCOVERY_ENTITIES
+    }
+
+
+def get_censored_entity_counts_for_scope(
```

Comment:
I would prefer (if we keep this utility, I'm not sure that it's needed - it should at the very least have a use in discovery api too) if this whole function was async, and then a sync version is created just for use in serializers.